### PR TITLE
Add version 2020-02-02-preview to  SecurityAlertPolicies APIs

### PR DESCRIPTION
**Pull request by ayeletshpigelman**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/202/Overall-Process-of-AME-Onboarding) for the workflow.
- [x] [Validation tools](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/84/Swagger-Validation-tools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/FAQ.md).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process]( https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review).

### CODE REVIEW

**Comment by ayeletshpigelman** [comment]
Hi
@qianwens can you please review?

**Comment by qianwens** [comment]
> Hi
> @qianwens can you please review?

Could you please help fix the lint failure? Add "WaitForARMFeedback" label since this is a new API version.

**Comment by ayeletshpigelman** [comment]
@qianwens the error is Operations API must be implemented for '/providers/Microsoft.Sql/operations'.
because this is new api version 2020-02-02-preview. I don't see this API in other versions.

**Comment by ayeletshpigelman** [comment]
@qianwens - Hi. any update?

**Comment by ayeletshpigelman** [comment]
@qianwens Hi. I see no progress with this PR.
can someone from ARM review it please?
The error in lint is: Operations API must be implemented for '/providers/Microsoft.Sql/operations'.
And I remember it can be ignored.
thanks,
Ayelet

**Comment by ayeletshpigelman** [comment]
@chiragg4u I added operations API and all tests pass. can you please review and let me know if you have any more comments?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/DatabaseSecurityAlertPolicies.json [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
```
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/DatabaseSecurityAlertPolicies.json:18]: creating a securityAlertPOlicy is possible via the PUT method, but there is no DELETE. Its typical to provide the delete option for something which is created by the user.

**Review by ravbhatnagar — COMMENTED** [review]
@ayeletshpigelman - one comment please take a look

File: specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/DatabaseSecurityAlertPolicies.json (hunk 1 above)
Comment (ravbhatnagar) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/DatabaseSecurityAlertPolicies.json:18]: after further discussion, it seems like this resource type is a singleton. the customer can either turn on this policy for a database or turn it off. Ideally, it should be modeled as a property on the database itself which can be updated via Patch.

Comment (ravbhatnagar) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/DatabaseSecurityAlertPolicies.json:18]: it seems like in previous api-versions, this already existed as a separate child. and so RP would like to keep it consistent with that. And also, the PUT is not really creating the resource. The resource exists in a disabled state even when the PUT hasnt happened. So, in addition to PUT, service team will also add a PATCH.

**Comment by ravbhatnagar** [comment]
notes - it seems like in previous api-versions, this already existed as a separate child. and so RP would like to keep it consistent with that. And also, the PUT is not really creating the resource. The resource exists in a disabled state even when the PUT hasnt happened. So, in addition to PUT, service team will create a workitem to add a PATCH. 
Signing off from ARM side.

**Review by qianwens — APPROVED** [review]

**Pull request merged**